v2/gcs-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/processing/handler/GCSToSourceStreamingHandler.java

```diff
@@ -89,6 +112,56 @@ public static String process(ProcessingContext taskContext, SpannerDao spannerDa
     return fileProcessedStartInterval;
   }
 
+  public static void writeFilteredEventsToGcs(
+      ProcessingContext taskContext,
+      Storage storage,
+      List<TrimmedShardedDataChangeRecord> filteredEvents) {
+    Matcher matcher = Pattern.compile(GCS_INPUT_DIRECTORY_REGEX).matcher(taskContext.getGCSPath());
```

Comment:
This can be done at the input level directly by using option level regexes - https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/f14c62cf269230ac12eafc0f551ea4ab17c07451/v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/options/SourceDbToSpannerOptions.java#L50

Reply:
Removed it. The parameter is of type @TemplateParameter.GcsReadFolder so the regex is already enforced